source/Plugins.cpp

```diff
@@ -345,6 +507,132 @@ const Set<Plugin> &Plugins::Get()
 // Toggles enabling or disabling a plugin for the next game restart.
 void Plugins::TogglePlugin(const string &name)
 {
+	lock_guard<mutex> guard(pluginsMutex);
 	auto *plugin = plugins.Get(name);
 	plugin->currentState = !plugin->currentState;
 }
+
+
+
+future<void> Plugins::Install(InstallData *installData, bool update)
```

Comment:
Copy the comment here from the header file.

Reply:
the comment in the header describes a block of functions, not specifically this one